blocks/main_loop.py

```diff
@@ -168,7 +168,7 @@ def run(self):
                         extension.main_loop = self
                     self._run_extensions('before_training')
                     with Timer('initialization', self.profile):
-                        self.algorithm.initialize()
+                        self.algorithm.initialize(profile=self.profile)
```

Comment:
One way to go would be to accept `theano_function_kwargs` dictionary in `GradientDescent.__init__` which would be further used in `initialize` when it is called in `run`. 

Another way to go would be to make `initialize` accept keyword arguments and have the user call it before `main_loop.run()`. But this requires a check for double initialization to be done inside `GradientDescent`. I would like to have it there instead of `MainLoop`, let alone for the reason that `blocks/algorithms.py` is almost a separate library, but it is not yet there. 

What do you think about the first proposal?